pkg/controllers/managementuser/rbac/project_handler.go

```diff
@@ -26,25 +27,81 @@ type pLifecycle struct {
 	m *manager
 }
 
+const updatePSAVerb = "updatepsa"
+
 func (p *pLifecycle) Create(project *v3.Project) (runtime.Object, error) {
 	for verb, suffix := range projectNSVerbToSuffix {
 		roleName := fmt.Sprintf(projectNSGetClusterRoleNameFmt, project.Name, suffix)
 		_, err := p.m.crLister.Get("", roleName)
+		// skip if clusterroles for project have already been created.
 		if err == nil || !apierrors.IsNotFound(err) {
 			continue
 		}
 
-		err = p.m.createProjectNSRole(roleName, verb, "", project.Name)
+		var psa bool
+		// here we ensure that the requisites to add updatepsa rule are met.
+		if suffix == projectNSVerbToSuffix[projectNSEditVerb] {
+			rt, err := p.rtLookup(project.Name)
+			if err != nil {
+				return project, err
+			}
+			psa = isUpdatepsaAllowed(rt, project.Name)
+		}
+
+		err = p.m.createProjectNSRole(roleName, verb, "", project.Name, psa)
 		if err != nil {
 			return project, err
 		}
-
 	}
 
 	err := p.ensureNamespacesAssigned(project)
 	return project, err
 }
 
+// rtLookup retrieves the roletemplate associated with the given project.
+func (p *pLifecycle) rtLookup(projectName string) (*v32.RoleTemplate, error) {
+	prtbs, err := p.m.prtbLister.List(projectName, labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+	if len(prtbs) == 0 {
+		return nil, fmt.Errorf("unable to list projectroletemplatebindings")
+	}
+	var rt string
+	for _, prtb := range prtbs {
+		// since prtbs are usually 2 (creator-project-owner and prtb-xxxxx),
+		// we just want the one that starts with "prtb-"
+		if prtb.Namespace == projectName && strings.HasPrefix(prtb.Name, "prtb-") {
```

Comment:
We can't assume `PRTBs` will always start with "prtb-" since they are part of the public API.
Additionally, a project can have many PRTBs which one should we check? (if we need to check any)

Reply:
> since they are part of the public API

Can you elaborate a bit more on this? It's not clear to me.

> a project can have many PRTBs which one should we check? (if we need to check any)

Based on my little experience observing the project management behavior, when a project is created a couple of new PRTBs are generated automatically:

* `creator-project-owner`
* `prtb-<random_string>`

We have to check the one with `prtb-` prefix, since is the one responsbile for binding the `project` with the `RoleTemplate` we are looking for.

How can a project have multiple prtbs? Isn't one prtb enough to specify the `roletemplate` we want to use to get permissions? Not sure we can use more than one `roletemplate` for project (at least from UI). Am I missing something?